pandas/plotting/_matplotlib/core.py

```diff
@@ -115,6 +116,67 @@ def __init__(
         self.sort_columns = sort_columns
 
         self.subplots = subplots
+        if not isinstance(self.subplots, (bool, Iterable)):
+            raise ValueError("subplots should be a bool or an iterable")
+        if isinstance(self.subplots, Iterable):
+
+            supported_kinds = (
+                "line",
+                "bar",
+                "barh",
+                "hist",
+                "kde",
+                "density",
+                "area",
+                "pie",
+            )
+            if self._kind not in supported_kinds:
+                raise ValueError(
+                    "When subplots is an iterable, kind must be "
+                    f"one of {', '.join(supported_kinds)}. Got {self._kind}"
+                )
+
+            if any(
+                not isinstance(group, Iterable) or isinstance(group, str)
```

Comment:
The thing is, we want `group` to be an iterable but we don't want `group` to be a str, which is also iterable.

Only using `if any(not isinstance(group, Iterable)` would allow strings and we don't want that